modules/internal/String.chpl

```diff
@@ -1121,20 +1121,22 @@ module String {
       var localThis: string = this.localize();
 
       var i = 0;
+      
+      
       while i < localThis.buffLen {
+        const (decodeRet, cp, nBytes) = decodeHelp(buff=localThis.buff, 
+                                                   buffLen=localThis.buffLen,
+                                                   offset=i, 
+                                                   allowEsc=true);
         const curPos = localThis.buff+i;
```

Comment:
`curPos` can also be removed, but it is not bad to keep it in

Reply:
`curPos` is there since it is passed as argument in bufferCopyLocal and it might look ugly to have aritmetic operation in parameters hence saved it. Rest can surely add it if you feel so.

Reply:
Yeah, I see. That's not a big worry. Just wanted to mention :)